### PR TITLE
Sort by all iteratees

### PR DESCRIPTION
**Pull request by megawac**
(I know this is the wrong file but I'm on a different computer :p)

Anyway @jdalton how do you feel about `sortByAll` and `sortByOrder` accepting more than just properties. I have a couple use cases for this namely multi column sorts with different importance factors.

As a bonus this should have deep support yeah?!!!

### CODE REVIEW

**Comment by jdalton** [comment]
Can you rebase this.

**Comment by megawac** [comment]
Yeah was more looking for thoughts before I fixed this up tmw morn

**Comment by jdalton** [comment]
I've known iteratees where coming :)

File: lodash.js [hunk 1]
```diff
@@ -2568,17 +2568,21 @@
...
      * @returns {Array} Returns the new sorted array.
      */
-    function baseSortByOrder(collection, props, orders) {
+    function baseSortByOrder(collection, comparitors, orders) {
       var index = -1,
           length = collection.length,
           result = isLength(length) ? Array(length) : [];
 
+      comparitors = arrayMap(comparitors, function(comparitor) {
+        return getCallback(comparitor, null, 1);
```
Comment (jdalton) [inline lodash.js]: `getCallback` can be hoisted out of `arrayMap` as `callback = getCallback()`.

Comment (jdalton) [inline lodash.js]: Can you add a unit test for these.

File: lodash.js [hunk 2]
```diff
@@ -2568,17 +2568,21 @@
...
+      });
+
       baseEach(collection, function(value) {
         var length = props.length,
             criteria = Array(length);
 
         while (length--) {
-          criteria[length] = value == null ? undefined : value[props[length]];
+          criteria[length] = comparitors[length](value);
         }
```
Comment (jdalton) [inline lodash.js]: The one thing that bothers me is there's no `this` binding. It wouldn't be possible. This makes it a bit awkward compared to `_.sortBy`.

Comment (megawac) [inline lodash.js]: Yep, that was the other thing - should context param be added to
sorybyorder and sortByAll?

Comment (jdalton) [inline lodash.js]: > Yep, that was the other thing - should context param be added

The thing is both methods can accept multiple prop names (potential iteratees) so the binding would apply to all of them? That gets messy. I think that using `_.bind` in this case would be better.

Comment (megawac) [inline lodash.js]: Works for me

File: lodash.js [hunk 3]
```diff
@@ -2568,17 +2568,21 @@
...
+    function baseSortByOrder(collection, comparitors, orders) {
       var index = -1,
           length = collection.length,
           result = isLength(length) ? Array(length) : [];
 
+      comparitors = arrayMap(comparitors, function(comparitor) {
+        return getCallback(comparitor, null, 1);
+      });
+
       baseEach(collection, function(value) {
```
Comment (megawac) [inline lodash.js]: @jdalton any reason you use `baseEach` instead of `baseMap` here?

File: lodash.js (hunk 1 above)
Comment (megawac) [inline lodash.js]: Will do

Comment (megawac) [inline lodash.js]: I guess the `1` doesn't matter here as theres no context

File: lodash.js (hunk 3 above)
Comment (jdalton) [inline lodash.js]: Nothing that can't be tweaked. `baseMap` would work.

**Comment by megawac** [comment]
Going to write up some tests and docs now - this implementation look good?

**Comment by jdalton** [comment]
Yap!

**Pull request merged**